[IMP] server: refactor ast_indexes into new node_index support from Ruff

fda-odoo opened the pull request:
Drop our home-made ast_indexes that help us indexing statements and use built-in node_indexes that are now directly on Nodes in last version of Ruff_python_ast.

mmahrouss commented on server/src/core/symbols/function_symbol.rs:
```diff
@@ -34,7 +34,7 @@ pub struct FunctionSymbol {
     pub is_property: bool,
     pub doc_string: Option<String>,
-    pub ast_indexes: Vec<u16>, //list of index to reach the corresponding ast node from file ast
+    pub ast_indexes: AtomicNodeIndex,
```
Suggestion: Maybe we could rename it to node_index? since it is just one index, not a sequence of indexes? also at symbol, class symbol. variable symbol ... ?

fda-odoo committed:
[IMP] server: refactor ast_indexes into new node_index support from Ruff

Drop our home-made ast_indexes that help us indexing statements and use built-in node_indexes that are now directly on Nodes in last version of Ruff_python_ast.
